engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/modules/output/table/html/helper/HtmlTextExtractorHelper.java

```diff
@@ -544,6 +555,28 @@ private boolean tryHandleDrawable( final ReportAttributeMap attrs, final long wi
     return true;
   }
 
+  private void convertImageDataToBase64( String name, AttributeList attrList, final ImageContainer image, final String type,
+    final float quality ) throws ContentIOException {
+    int indexStart = name.lastIndexOf( "picture" );
+    String imgAltText = name.substring( indexStart );
+    attrList.setAttribute( HtmlPrinter.XHTML_NAMESPACE, ALT_ATTR, imgAltText );
+    attrList.setAttribute( HtmlPrinter.XHTML_NAMESPACE, SRC_ATTR, "poc-src-placeholder-" + imgAltText );
+    try {
+      final DefaultHtmlContentGenerator.ImageData imageData =
+        contentGenerator.getImageData( image, type, quality, true );
+      final String HTML_IMG_CSS_BASE64_FORMAT = "url('data:%1$s;base64,%2$s')";
+      String urlBase64Encoding = String.format( HTML_IMG_CSS_BASE64_FORMAT,
+        imageData.getMimeType(), getBase64( imageData ) );
+      attrList.setAttribute( HtmlPrinter.XHTML_NAMESPACE, "content", urlBase64Encoding );
```

Comment:
string "content" needs to be replaced by newly added StyleBuilder.java enums  `CSSKeys#CONTENT` directly or `CSSKeys#CONTENT#getCssName()`

Reply:
addressed